pkg/endpoint/policy.go

```diff
@@ -155,87 +155,143 @@ func getSecurityIdentities(owner Owner, selector *api.EndpointSelector) []policy
 	return identities
 }
 
-func (e *Endpoint) removeOldFilter(owner Owner, filter *policy.L4Filter) int {
+// removeOldFilter removes the old l4 filter from the endpoint and returns
+// a map of the L3-dependent L4 rules that were successfully removed if the
+// value is true, or in case of an error the value is false.
+func (e *Endpoint) removeOldFilter(owner Owner, filter *policy.L4Filter) (map[policy.NIPortProto]bool, int) {
+	fromEndpointsSrcIDs := map[policy.NIPortProto]bool{}
 	port := uint16(filter.Port)
 	proto, err := u8proto.ParseProtocol(string(filter.Protocol))
 	if err != nil {
 		e.getLogger().WithError(err).Warn("Parse policy protocol failed")
-		return 1
+		return fromEndpointsSrcIDs, 1
 	}
 
 	errors := 0
 	for _, sel := range filter.FromEndpoints {
 		for _, id := range getSecurityIdentities(owner, &sel) {
 			srcID := id.Uint32()
+			nipp := policy.NIPortProto{
+				SecID: id,
+				Port:  port,
+				Proto: uint8(proto),
+			}
 			if err = e.PolicyMap.DeleteL4(srcID, port, uint8(proto)); err != nil {
 				e.getLogger().WithError(err).WithField(logfields.L4PolicyID, srcID).Debug("Delete old l4 policy failed")
+				fromEndpointsSrcIDs[nipp] = false
+			} else {
+				fromEndpointsSrcIDs[nipp] = true
 			}
 		}
 	}
 
-	return errors
+	return fromEndpointsSrcIDs, errors
 }
 
-func (e *Endpoint) applyNewFilter(owner Owner, filter *policy.L4Filter) int {
+// applyNewFilter adds the given l4 filter to the endpoint and returns a map of
+// the L3-dependent L4 rules that were successfully removed if the
+// value is true, or in case of an error the value is false.
```

Comment:
Please specify what "if the value is true" means; the value of what?

Reply:
Fixed